spinoso-string/src/enc/mod.rs

```diff
@@ -69,7 +69,7 @@ impl PartialEq for EncodedString {
         // [3.0.2] > s == t
         // => true
         // ```
-        *self.as_slice() == *other.as_slice()
+        *self.as_slice() == *other.as_slice() && self.char_len() == other.char_len()
```

Comment:
this test can be made cheaper by comparing the encodings I think, but this works for now.

Reply:
```suggestion
        (*self.as_slice() == *other.as_slice() && self.encoding() == other.encoding()) ||
            (*self.as_slice() == * other.as_slice() && self.encoding() != self.encoding && self.is_ascii_only() && other.is_ascii_only())
```

:point_up: I didn't test this, so it's likely invalid syntax, however I think this is valid for at least the 3 encodings supported by `spinoso-string`

```console
irb(main):001:0> a = "\x7f"
=> "\u007F"
irb(main):002:0> b = a.b
=> "\x7F"
irb(main):003:0> a.encoding
=> #<Encoding:UTF-8>
irb(main):004:0> b.encoding
=> #<Encoding:ASCII-8BIT>
irb(main):005:0> a == b
=> true
irb(main):006:0> c = "\x80"
=> "\x80"
irb(main):007:0> d = c.b
=> "\x80"
irb(main):008:0> c == d
=> false
```

Reply:
@b-n You meant `self.encoding() != other.encoding()` not `self.encoding() != self.encoding` right?

Reply:
Ah sorry, missed this question. I will reformat my suggestion since it's hard to see what it was trying to do :grimacing: 

```suggestion
        if self.encoding() == other.encoding() {
            *self.as_slice() == *other.as_slice()
        } else {
            *self_as_slice() == *other.as_slice() && self.is_ascii_only() && other.is_ascii_only()
        }
```

I noticed that if two ruby strings contain exactly the same characters, and so long as they are only ascii characters (0..=127) then it says it's the "same", even if the encoding doesn't match.

Reply:
@b-n I've made this change. let me know what you think

Reply:
Rust match block++, very very very nice.